doc/source/cluster/kubernetes/k8s-ecosystem/yunikorn.md

```diff
@@ -0,0 +1,184 @@
+
+KubeRay's Apache YuniKorn integration enables more efficient scheduling of Ray Pods in multi-tenant Kubernetes environments.
+
+:::{note}
+
+This feature requires KubeRay version 1.2.2 or newer, and it's still alpha.
+
+:::
+
+### Step 1: Create a Kubernetes cluster with KinD
+Run the following command in a terminal:
+
+```shell
+kind create cluster
+```
+
+### Step 2: Install Apache YuniKorn
+
+You need to successfully install Apache YuniKorn on your Kubernetes cluster before enabling Apache YuniKorn integration with KubeRay.
+See [Get Started](https://yunikorn.apache.org/docs/) for Apache YuniKorn installation instructions.
+
+### Step 3: Install the KubeRay operator with Apache YuniKorn support
+
+When installing KubeRay operator using Helm, you should pass the `--set batchScheduler.name=yunikorn` flag when running on the command line:
+
+```shell
+helm install kuberay-operator kuberay/kuberay-operator --version 1.2.2 --set batchScheduler.name=yunikorn
+```
+
+### Step 4: Use Apache YuniKorn for gang scheduling
+
+This example walks through how gang scheduling works with Apache YuniKorn and KubeRay.
+
+First, create a queue with a capacity of 4 CPUs and 6Gi of RAM by editing the ConfigMap:
+
+Run `kubectl edit configmap -n yunikorn yunikorn-defaults`
+
+This `ConfigMap` is created during the installation of the Apache YuniKorn Helm chart.
+
+Add a `queues.yaml` config to under the `data` key, the final ConfigMap should look like this:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  # Metadata for the ConfigMap, skip for brevity
+data:
+  queues.yaml: |
+    partitions:
+      - name: default
+        queues:
+          - name: root
+            queues:
+              - name: test
+                submitacl: "*"
+                parent: false
+                resources:
+                  guaranteed:
+                    memory: 6G
+                    vcore: 4
+                  max:
+                    memory: 6G
+                    vcore: 4
+```
+
+Save the changes and exit the editor. This configuration creates a queue named `root.test` with a capacity of 4 CPUs and 6Gi of RAM.
+
+Next, create a RayCluster with a head node (1 CPU + 2Gi of RAM) and two workers (1 CPU + 1Gi of RAM each), for a total of 3 CPU and 4Gi of RAM:
+
+```shell
+# Path: kuberay/ray-operator/config/samples
+# Configure the necessary labels on the RayCluster custom resource for Apache YuniKorn scheduler's gang scheduling:
+# - `ray.io/gang-scheduling-enabled`: This should be set to `true` to enable gang scheduling.
+# - `yunikorn.apache.org/app-id`: This should be set to the name of the RayCluster.
```

Comment:
It is OK to set with the name of the RayCluster, but since this will be used as the indentifier as the app in the scheduler, we should explicitly say make sure the name doesn't conflict with others existing Ray clusters (even in other namespace). One caveat is, we can have the same RayCluster name in different namespaces, but if that same name is used, these RayClusters will be considered as belonging to the same application in the yunikorn scheduler.

Reply:
done